src/utils/anonymousUserEventManager.ts

```diff
@@ -0,0 +1,356 @@
+          method: 'POST',
+          url: ENDPOINT_TRACK_ANON_SESSION,
+          data: payload
+        }).catch((e) => {
+          if (e?.response?.status === 409) {
+            this.getAnonCriteria();
+          }
+        });
+        if (response && response.status === 200) {
+          await setAnonUserId(userId);
+          this.syncEvents();
+        }
+      }, 500);
+    }
+  }
+
+  async syncEvents() {
+    const strTrackEventList = localStorage.getItem(SHARED_PREFS_EVENT_LIST_KEY);
+    const trackEventList = strTrackEventList
+      ? JSON.parse(strTrackEventList)
+      : [];
+
+    if (trackEventList.length) {
+      for (let i = 0; i < trackEventList.length; i++) {
+        const event = trackEventList[i];
+        const eventType = event[SHARED_PREFS_EVENT_TYPE];
+        delete event.eventType;
+        switch (eventType) {
+          case TRACK_EVENT: {
+            this.track(event);
+            break;
+          }
+          case TRACK_PURCHASE: {
+            this.trackPurchase(event);
+            break;
+          }
+          case TRACK_UPDATE_CART: {
+            this.updateCart(event);
+            break;
+          }
+          case UPDATE_USER: {
+            this.updateUser({ dataFields: event });
+            break;
+          }
+          default:
+            break;
+        }
+
+        localStorage.removeItem(SHARED_PREFS_ANON_SESSIONS);
+        localStorage.removeItem(SHARED_PREFS_EVENT_LIST_KEY);
+      }
+    }
+  }
+
+  private async storeEventListToLocalStorage(
+    newDataObject: any,
+    shouldOverWrite: boolean
+  ) {
+    const strTrackEventList = localStorage.getItem(SHARED_PREFS_EVENT_LIST_KEY);
+    let previousDataArray = [];
+
+    if (strTrackEventList) {
+      previousDataArray = JSON.parse(strTrackEventList);
+    }
+
+    if (shouldOverWrite) {
+      const trackingType = newDataObject[SHARED_PREFS_EVENT_TYPE];
+      const indexToUpdate = previousDataArray.findIndex(
+        (obj: any) => obj[SHARED_PREFS_EVENT_TYPE] === trackingType
+      );
+      if (indexToUpdate !== -1) {
+        const dataToUpdate = previousDataArray[indexToUpdate];
+
+        previousDataArray[indexToUpdate] = {
+          ...dataToUpdate,
+          ...newDataObject
+        };
+      } else {
+        previousDataArray.push(newDataObject);
+      }
+    } else {
+      previousDataArray.push(newDataObject);
+    }
+
+    localStorage.setItem(
+      SHARED_PREFS_EVENT_LIST_KEY,
+      JSON.stringify(previousDataArray)
+    );
+    const criteriaId = await this.checkCriteriaCompletion();
+    if (criteriaId !== null) {
+      this.createKnownUser(criteriaId);
+    }
+  }
+
+  private getCurrentTime = () => {
+    return new Date().getTime();
+  };
+
+  private getWebPushOptnIn(): string {
+    const notificationManager = window.Notification;
+    if (notificationManager && notificationManager.permission === 'granted') {
+      return window.location.hostname;
+    } else {
+      return '';
+    }
+  }
```

Comment:
web push??

Reply:
yes we are supposed to return hostname here if web push is enabled in `anonymoususer/events/session` API